BaragonAgentService/src/test/java/com/hubspot/baragon/agent/preferSameRackWeightingBalancedTest.java

```diff
@@ -0,0 +1,106 @@
+package com.hubspot.baragon.agent;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.jukito.JukitoRunner;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Multiset;
+import com.hubspot.baragon.agent.config.BaragonAgentConfiguration;
+import com.hubspot.baragon.agent.handlebars.PreferSameRackWeightingHelper;
+import com.hubspot.baragon.models.BaragonAgentEc2Metadata;
+import com.hubspot.baragon.models.BaragonAgentMetadata;
+import com.hubspot.baragon.models.UpstreamInfo;
+
+
+@RunWith(JukitoRunner.class)
+public class preferSameRackWeightingBalancedTest {
+
+  final List<String> availabilityZones = Arrays.asList("us-east-1", "us-east-1", "us-east-2", "us-east-3", "us-east-5");
```

Comment:
For consistency with what we'll actually see, these values will be more like `us-east-1a, us-east-1b, us-east-1e`. us-east-(1,2) are different aws 'regions' while the availability zones are the a,b,etc within a region